core/sql/qmscommon/QRLogger.cpp

```diff
@@ -186,6 +188,9 @@ NABoolean QRLogger::initLog4cxx(const char* configFileName)
 {
   NAString logFileName;
 
+  if (gv_QRLoggerInitialized_)
+     return TRUE;
+ 
   // get the log directory
```

Comment:
The caller sqInit is protected by a semaphore.  sqInit is called by the first CLI call in a process . See the function below in the same file CliExtern.cpp

908 static Lng32 CliNonPrivPrologue()
    909 {
    910 
    911   Lng32 retcode;
    912 
    913   if (cli_globals == NULL || cli_globals->getIsInitialized() == FALSE)
    914   {
    915     globalSemaphore.get();
    916     if (cli_globals != NULL)
    917     {
    918       globalSemaphore.release();
    919       return 0;
    920     }
    921     sqInit();
    922     CliGlobals::createCliGlobals(FALSE); // this call will create the globals.
    923     ex_assert(SQL_EXEC_SetEnviron_Internal(1) == 0, "Unable to set the Environment");
    924     globalSemaphore.release();
    925   }
    926   return 0;
    927 }